- Added Trim Functions into the ShooterWheelSubsystem

johnylamw opened the pull request:
- Added Trim Functions into the ShooterWheelSubsystem
- Added a Trim Function Test
- Added Two Instant Commands into OperatorCommandMap that needs controller assignment

johnylamw committed:
- Added Trim Functions into the ShooterWheelSubsystem
- Added a Trim Function Test
- Added Two Instant Commands into OperatorCommandMap that needs controller assignment

aschokking commented on src/main/java/competition/operator_interface/OperatorCommandMap.java:
```diff
@@ -81,6 +85,9 @@ public void setupShooterCommands(OperatorInterface operatorInterface, ShooterWhe
         operatorInterface.manualOperatorGamepad.getifAvailable(XboxButton.RightBumper).whenPressed(speedUp);
         operatorInterface.manualOperatorGamepad.getifAvailable(XboxButton.LeftBumper).whenPressed(slowDown);
+
+        // assign a button to increaseTrim
```
add TODO to the comment

JohnGilb commented on src/main/java/competition/operator_interface/OperatorCommandMap.java:
```diff
@@ -73,6 +73,10 @@ public void setupShooterCommands(OperatorInterface operatorInterface, ShooterWhe
         Command speedUp = new InstantCommand(() -> shooter.changeTargetRPM(500), shooter.getSetpointLock());
         Command slowDown = new InstantCommand(() -> shooter.changeTargetRPM(-500), shooter.getSetpointLock());
+
+        Command increaseTrim = new InstantCommand(() -> shooter.increaseTargetTrimRPM(100));
```
`changeTrimRPM()`

aschokking commented on src/main/java/competition/subsystems/shooterwheel/ShooterWheelSubsystem.java:
```diff
@@ -29,6 +31,7 @@ public ShooterWheelSubsystem(CommonLibFactory factory, PropertyFactory pf, Ideal
 
         targetRpmProp = pf.createEphemeralProperty("TargetRPM", 0);
         currentRpmProp = pf.createEphemeralProperty("CurrentRPM", 0);
+        rpmTrimProp = pf.createPersistentProperty("TrimRPM", 0);
```
change to ephemeral

JohnGilb replied:
`ephemeral` property

JohnGilb commented on src/main/java/competition/subsystems/shooterwheel/ShooterWheelSubsystem.java:
```diff
@@ -44,7 +47,27 @@ public void setTargetRPM(double speed) {
 
     public double getTargetRPM() {
-        return targetRpmProp.get();
+        if (targetRpmProp.get() == 0) {
```
remember, don't compare doubles to 0. You want something closer to:

`if (Math.abs(targetRpmProp.get()) < 50)`

johnylamw committed:
- renamed increaseTargetTrimRPM -> changeTrimRPM
- readjusted double property from persistent -> empheral
- changed getTargetRPM() to not trim if the targetRPM is less than 50

johnylamw committed:
-removed extra blank line :)

aschokking committed:
Merge branch 'master' into feature/ShotPowerTrimming

aschokking committed:
Merge branch 'master' into feature/ShotPowerTrimming
